modules/9_export/export.qmd

```diff
@@ -122,173 +125,201 @@ if(!file.exists(prev_sc_obj)) {
+
+**Notes**: 
+- Due to license issues, this is only supported for 10x datasets. Non-10x datasets will not be exported.
+- Spatial information is not included in the cloupe file. For 10x Visium/VisiumHD data, we recommend to use the original cloupe file provided by Spaceranger and to import the CSV cell metadata (see above for details) in Loupe. For 10x Xenium, the cloupe file can be loaded into the Loupe Cell Browser to inspect the cell clustering. For spatial information, use the Xenium explorer (see below for details).
+
+```{r}
+#| label: create_loupe_file
+#| results: asis
+
+# What do we export
+create_loupe_file = param("create_loupe_file")
+
+# Create a cloupe file for the Loupe Browser (only 10x, 10x_visium and 10x_xenium)
+if (create_loupe_file & all(grepl("10x", sc$technology))) {
+  dir.create(file.path(module_dir, "results", "loupe"), showWarnings=FALSE, recursive=TRUE)
+  
+  # Assays to export (currently only default)
+  assays_to_export = SeuratObject::DefaultAssay(sc)
+  
+  # Cell metadata to export (ignore categories based on sketch assay)
+  # Note: Loupe does not support numeric columns
+  categories_to_export = colnames(sc[[]])
+  categories_to_export = categories_to_export[!grepl("\\.sketch$", categories_to_export)]
+  categories_to_export = categories_to_export[categories_to_export != "orig_barcode"]
+  
+  # Embeddings to export (ignore embeddings that were done based on sketch assays)
+  embeddings_to_export = purrr::discard(SeuratObject::Reductions(sc), function(r) {
+    return(grepl("\\.sketch$", Seurat::DefaultAssay(sc[[r]])))
+  })
+  
+  # Export as loupe file
+  ExportLoupe(sc, 
+              assays=assays_to_export, 
+              categories=categories_to_export, 
+              embeddings=embeddings_to_export, 
+              output_dir=file.path(module_dir, "results", "loupe"),
+              output_name="scrnaseq2")
+  
+  CalloutBox("The following loupe file can be loaded directly into the loupe browser: 'loupe/scrnaseq2.cloupe'.", type="note")
+} else {
+  CalloutBox("Creating loupe files was deactivated or is supported only for datasets of technology '10x', '10x_xenium' or '10x_visium'.", type="note")
+}
+```
+
+The Xenium Explorer is a tool to explore 10x Xenium spatial data in a variety of ways. For download, please visit the [Xenium Explorer website](https://www.10xgenomics.com/support/software/xenium-explorer/latest). There are also a number of tutorials available on the [website](https://www.10xgenomics.com/support/software/xenium-explorer/latest/tutorials).
+
+For 10x Xenium data, categorial cell metadata is exported as analysis.zarr.zip file for each dataset. Replace the original analysis.zarr.zip in the respective Xenium dataset (make a backup) to include the results of this analysis in the Xenium explorer. 
+
+```{r}
```

Comment:
Always ask in header if spatial data is applicable (for rendering); and check again in the chunk (for running in R); apply for all spatial chunks.

Reply:
Done